### PR TITLE
PS-5263: no tmpdir space can lead to assert in THD::send_statement_status (8.0)

### PR DESCRIPTION
**Pull request by inikep**
1. Update `handle_binlog_flush_or_sync_error()` to set `my_ok(thd)` after `thd->clear_error()`.
2. Update `rpl_bug72457.inc` to support errorless statements that disable binary logging.
3. Re-record `rpl_bug72457_innodb` and `rpl_bug72457_myisam`.

### CODE REVIEW

**Comment by inikep** [comment]
Tests started at https://ps.cd.percona.com/view/8.0/job/percona-server-8.0-param/137/

File: mysql-test/suite/rpl/t/rpl_bug72457_innodb-master.opt [hunk 1]
```diff
@@ -1,2 +1,3 @@
 --internal-tmp-disk-storage-engine=InnoDB
 --default-tmp-storage-engine=InnoDB
+--log-error=$MYSQLTEST_VARDIR/tmp/rpl_bug72457.err
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rpl/t/rpl_bug72457_innodb-master.opt]: This is still not good enough in the case of MTR --repeat: the testcase should restart the server in the .test file itself with the custom path, then restart in the end with default, remove the custom error log

**Review by laurynas-biveinis — CHANGES_REQUESTED** [review]

**Comment by inikep** [comment]
Tested at https://ps.cd.percona.com/view/8.0/job/percona-server-8.0-param/181/

**Review by laurynas-biveinis — APPROVED** [review]

**Comment by inikep** [comment]
Tested at https://ps.cd.percona.com/view/8.0/job/percona-server-8.0-param/190/

**Pull request merged**